lively.morphic/world-loading.js

```diff
@@ -312,12 +313,31 @@ export async function interactivelySaveWorld (world, options) {
         const { author: { name: authorName }, timestamp } = newerCommit;
         const overwriteQ = `The current version of world ${name} is not the most recent!\n` +
                        `A newer version by ${authorName} was saved on ` +
-                       `${date.format(new Date(timestamp), 'yyyy-mm-dd HH:MM')}. Overwrite?`;
-        overwrite = await world.confirm(['Version Conflict\n', null, overwriteQ, { fontSize: 16, fontWeight: 'normal' }], { width: 600 });
+                       `${date.format(new Date(timestamp), 'yyyy-mm-dd HH:MM')}. Do you want to merge the worlds?`;
+        overwrite = await world.multipleChoicePrompt(['Version Conflict\n', null, overwriteQ, { fontSize: 16, fontWeight: 'normal' }], { width: 600, choices: ['Take mine', 'Merge mine', 'Manual merge', 'Merge theirs', 'Take theirs'] });
       }
-      if (!overwrite) return null;
-      world.changeMetaData('commit', obj.dissoc(newerCommit, ['preview']), /* serialize = */true, /* merge = */false);
-      return interactivelySaveWorld(world, { ...options, morphicdb: db, showSaveDialog: false });
+      let result = null;
+      switch (overwrite) {
+        case 'Take mine':
+          world.changeMetaData('commit', obj.dissoc(newerCommit, ['preview']), /* serialize = */true, /* merge = */false);
+          result = interactivelySaveWorld(world, { ...options, morphicdb: db, showSaveDialog: false });
+          break;
+        case 'Merge mine':
+          // todo load the new version, merge their changes if merge conflict take mine
+          break;
+        case 'Manual merge':
+          // todo
+          break;
+        case 'Merge theirs':
+          // todo load the actual version, merge my changes if merge conflict take theirs
+          break;
```

Comment:
Can't we already call the world merging function even if it doesn't do anything yet?n

```suggestion
        case 'Merge mine':
        case 'Manual merge':
        case 'Merge theirs':
          mergeWorlds(expectedVersion, actualVersion, strategy);
          break;
```